de.dlr.sc.virsat.model.test/src/de/dlr/sc/virsat/model/dvlm/roles/RightsHelperTest.java

```diff
@@ -95,14 +95,16 @@ public void testHasSystemUserWritePermission() {
 		assertTrue("Yes we can write to the object", RightsHelper.hasSystemUserWritePermission(ca));
 		assertTrue("Yes we can write to the object", RightsHelper.hasSystemUserWritePermission(vpi));
 
-		discipline.setUser("TheBadGuy");
-
-		assertFalse("No we cannot write to the object", RightsHelper.hasSystemUserWritePermission(sei));
+		discipline.getUsers().add("TheBadGuy");
 		
-		assertFalse("No we cannot write to the object", RightsHelper.hasSystemUserWritePermission(ca));
+		assertTrue("No we cannot write to the object", RightsHelper.hasSystemUserWritePermission(sei));

```

Comment:
Don't you change the complete test behavior here? I think the idea of the test was to ensuere that on the one side, TestUser has write permissions and on the other side "TheBadGuy" doesn't have write permissions... You simply change the asserts to true which makes the test complete, but doesn't make sense anymore. I think to keep the original concept of the test, you could remove the TestUser and make sure if there is just the TheBadGuy in the discipline there are no rights granted...

Reply:
Also the write permission changes from ca to a sei?

Reply:
Also the object to be checked changes from ca to sei? But I guess that does not make a differrence?

Reply:
Ok, looks god now! Only thing left: the comment is missleading now. In L100 we assertTrue but the comment mentions "we cannot write"... Which is outdated... Please update this so that other new people to Virtual Satellite can understand what the test does :)